event-adapter/pkg/apiserver/installer/apiserver_test.go

```diff
@@ -130,170 +113,51 @@ func handle(prov provider.CustomMetricsProvider) http.Handler {
-				trimmedValues.Items = append(trimmedValues.Items, custom_metrics.MetricValue{})
-			}
-		}
-	} else {
-		trimmedValues = custom_metrics.MetricValueList{
-			Items: values,
-		}
-	}
-
-	return &trimmedValues, nil
-}
-
-func (p *fakeProvider) GetNamespacedMetricByName(groupResource schema.GroupResource, namespace string, name string, metricName string) (*custom_metrics.MetricValue, error) {
-	metricId := namespace+"/"+groupResource.String()+"/"+name+"/"+metricName
-	values, ok := p.namespacedValues[metricId]
-	if !ok {
-		return nil, fmt.Errorf("non-existant metric requested (id: %s)", metricId)
-	}
+type fakeProvider struct{}
 
-	return &values[0], nil
+func (p *fakeProvider) GetNamespacedEventsByName(namespace, name string) (*types.EventValue, error) {
+	return &types.EventValue{}, nil
 }
 
-func (p *fakeProvider) GetNamespacedMetricBySelector(groupResource schema.GroupResource, namespace string, selector labels.Selector, metricName string) (*custom_metrics.MetricValueList, error) {
-	metricId := namespace+"/"+groupResource.String()+"/*/"+metricName
-	values, ok := p.namespacedValues[metricId]
-	if !ok {
-		return nil, fmt.Errorf("non-existant metric requested (id: %s)", metricId)
-	}
-
-	var trimmedValues custom_metrics.MetricValueList
-
-	if trimmedCount, ok := p.namespacedSubsetCounts[metricId]; ok {
-		trimmedValues = custom_metrics.MetricValueList{
-			Items: make([]custom_metrics.MetricValue, 0, trimmedCount),
-		}
-		for i := range values {
-			var lbls labels.Labels
-			if i < trimmedCount {
-				lbls = matchingSet
-			} else {
-				lbls = emptySet
-			}
-			if selector.Matches(lbls) {
-				trimmedValues.Items = append(trimmedValues.Items, custom_metrics.MetricValue{})
-			}
-		}
-	} else {
-		trimmedValues = custom_metrics.MetricValueList{
-			Items: values,
-		}
-	}
-
-	return &trimmedValues, nil
-}
-
-func (p *fakeProvider) ListAllMetrics() []provider.MetricInfo {
-	return p.metrics
+func (p *fakeProvider) ListAllEventsByNamespace(namespace string) (*types.EventValueList, error) {
+	return &types.EventValueList{}, nil
 }
 
-func TestCustomMetricsAPI(t *testing.T) {
-	totalNodesCount := 4
-	totalPodsCount := 16
-	matchingNodesCount := 2
-	matchingPodsCount := 8
-
+func TestEventsAPI(t *testing.T) {
 	type T struct {
-		Method        string
-		Path          string
-		Status        int
-		ExpectedCount int
+		Method string
+		Path   string
+		Status int
 	}
-	cases := map[string]T{
-		// checks which should fail
-		"GET long prefix": {"GET", "/" + prefix + "/", http.StatusNotFound, 0},
-
-		"root GET missing storage":    {"GET", "/" + prefix + "/" + groupVersion.Group + "/" + groupVersion.Version + "/blah", http.StatusNotFound, 0},
 
-		"namespaced GET long prefix":        {"GET", "/" + prefix + "/", http.StatusNotFound, 0},
-		"namespaced GET missing storage":    {"GET", "/" + prefix + "/" + groupVersion.Group + "/" + groupVersion.Version + "/blah", http.StatusNotFound, 0},
+	group := "v1events/v1alpha1"
+	cases := map[string]T{
```

Comment:
Couple of comments about your test cases:
- "GET list of events (singular)" - it's not important that you use singular form of "events", it's important that you use any word other than "events". I would change comment to "GET list of unknown objects" and path to "/namespaces/default/foo" for clarity
- "GET evento" - exactly the same - my suggestion is "GET unknown object by name" and "/namespaces/default/foo/bar"
- "GET wrong prefix (singular)" - it doesn't matter if it's v1event or v1foo - what about just "GET wrong prefix" and "//apis/v1foo/v1alpha1/"?
- What about other test cases? What should happen for paths "/namespaces/default", "/namespaces/*/events", "/namespaces/default/events/my/long/event"?
- Why are you expecting StatusInternalError? Your server shouldn't return internal error when the path is wrong, StatusNotFound seems more appropriate here. I don't require you to fix it in this PR, but at least add a TODO saying that it's not working as intended (or explain it if it does).